### PR TITLE
LVM2 support and minor tidying

### PR DESCRIPTION
**Pull request by TimJDFletcher**
I have fixed up the script to support LVM devices using wait-for-root from initramfs, as well as tweaking the sed match to find lvm root devices.

I have also put some minor corrections in the flags calling fsck and resize2fs

I've tested my changes on a LVM configured VM

### CODE REVIEW

File: debian-resizefs.sh [hunk 1]
```diff
@@ -5,7 +5,7 @@
 # Alternative: http://www.ivarch.com/blogs/oss/2007/01/resize-a-live-root-fs-a-howto.shtml
 
 # Check current filesystem type
-ROOT_FS_TYPE="$(sed -n -e 's|^/dev/[a-z]\+[1-9]\+ / \(ext4\) .*$|\1|p' /proc/mounts)"
+ROOT_FS_TYPE="$(sed -n -e 's|^/dev/.* / \(ext4\) .*$|\1|p' /proc/mounts)"
```
Comment (szepeviktor) [inline debian-resizefs.sh]: 👍

File: debian-resizefs.sh [hunk 2]
```diff
@@ -48,11 +48,9 @@ case "$1" in
...
 esac
 
-/sbin/e2fsck -f "$ROOT" || echo "e2fsck: $?"
-# Size in filesystem blocks, usually 4 KB
-# tune2fs -l /dev/vda1
-# 1310720 blocks = 5 GB
-/sbin/resize2fs -d 8 "$ROOT" 1310720 || echo "resize2fs: $?"
+wait-for-root "${ROOT}" 20
+/sbin/e2fsck -y -f "$ROOT" || echo "e2fsck: $?"
+/sbin/resize2fs -d 8 "$ROOT" 8G || echo "resize2fs: $?"
```
Comment (szepeviktor) [inline debian-resizefs.sh]: Could you add a comment to tell that "8G" should be edited?

**Comment by szepeviktor** [comment]
I've never thought some else is looking at my code!

File: debian-resizefs.sh (hunk 1 above)
Comment (szepeviktor) [reply debian-resizefs.sh]: In `[1-9]\+` what is the purpose of `\+`? Make up numbers from digits 1-9?

Comment (TimJDFletcher) [reply debian-resizefs.sh]: I'm not great with sed but I think that is a joiner ie the match is exactly:

/dev/[any letters]+[followed by number]+followed by a space

File: debian-resizefs.sh (hunk 2 above)
Comment (TimJDFletcher) [reply debian-resizefs.sh]: Switched it to a var and added a comment

**Comment by TimJDFletcher** [comment]
I was looking for how to shrink the rootfs for a cloud VM I was taking a template of as the VM had lots of wasted space in root and I found your code. It was very helpful and I think I'll build and ansible role from it to put in our build pipelines.

File: debian-resizefs.sh (hunk 1 above)
Comment (szepeviktor) [reply debian-resizefs.sh]: I think it is enough to support /dev/sda0-/dev/sda9, so without `\+`

File: debian-resizefs.sh (hunk 2 above)
Comment (szepeviktor) [reply debian-resizefs.sh]: OK!

File: debian-resizefs.sh (hunk 1 above)
Comment (TimJDFletcher) [reply debian-resizefs.sh]: The breaks for LVM root volumes eg: 
```
cat /proc/mounts | grep ' / '
/dev/mapper/vg-root / ext4 rw,relatime,errors=remount-ro,data=ordered 0 0
```

Comment (szepeviktor) [reply debian-resizefs.sh]: Excuse me! All the way I though `.*` is mine!

Comment (szepeviktor) [reply debian-resizefs.sh]: Maybe `\S\+` would have been the best!

**Pull request merged**